core/frontend/src/store/settings.ts

```diff
@@ -20,107 +25,125 @@ class SettingsStore extends VuexModule {
 
   is_pirate_mode = false
 
+  // Secret mode for developers that know our secret.
+  // It's enabled when the vehicle name is 'Sir Francis Drake', one the most famous pirates,
+  // and the pirate mode toggled to be enabled.
+  is_dev_mode = false
+
   last_version_update_notification_time = 0 // Start in 1970: https://www.youtube.com/watch?v=wwcKs5K1oWg
 
   tour_version = 0
 
   user_top_widgets = [] as string[]
 
-    @Mutation
+  @Mutation
   setDarkTheme(value: boolean): void {
     this.is_dark_theme = value
     SettingsStore.save()
   }
 
-    @Mutation
-    setPirateMode(value: boolean): void {
-      this.is_pirate_mode = value
-      SettingsStore.save()
-    }
+  @Mutation
+  setPirateMode(value: boolean): void {
+    this.is_pirate_mode = value
+    SettingsStore.checkDevMode()
+    SettingsStore.save()
+  }
 
-    @Mutation
-    updateVersionUpdateNotificationTime(): void {
-      this.last_version_update_notification_time = new Date().getTime()
-      SettingsStore.save()
-    }
+  @Mutation
+  updateVersionUpdateNotificationTime(): void {
+    this.last_version_update_notification_time = new Date().getTime()
+    SettingsStore.save()
+  }
 
-    @Mutation
-    setTourVersion(value: number): void {
-      this.tour_version = value
-      SettingsStore.save()
-    }
+  @Mutation
+  setTourVersion(value: number): void {
+    this.tour_version = value
+    SettingsStore.save()
+  }
 
-    @Mutation
-    setTopWidgets(widgets: string[]): void {
-      this.user_top_widgets = widgets
-      SettingsStore.save()
-    }
+  @Mutation
+  setTopWidgets(widgets: string[]): void {
+    this.user_top_widgets = widgets
+    SettingsStore.save()
+  }
 
-    /**
-     * Get name of the settings variable on the system
-     * @param string name
-     * @returns string
-     */
-    private static settingsName(name: string): string {
-      return `bluerobotics-blueos-${name}`
-    }
+  static checkDevMode(): void {
+    const { is_pirate_mode } = SettingsStore.state
+    const is_dev_mode = beacon.vehicle_name === 'Sir Francis Drake' && is_pirate_mode
+    Vue.set(SettingsStore.state, 'is_dev_mode', is_dev_mode)
```

Comment:
I wonder if we could use a [getter](https://vuex.vuejs.org/guide/getters.html) to make sure this is reactive regarding the vehicle name.  They seem to fit the use case, but I'm not sure if we could run into issues as the name is in a different store

Reply:
you cant have a computed or a getter from vuex decorators from what I searched for.